File: Agents/Azure-Agent-Skills-In-Action/skill-demos/sdk-python/05_azure_search_documents.py
"""
TRIPLE:
  Skill: azure-search-documents-py
  Prompt: "Using azure-search-documents-py skill, write a Python script that performs
           hybrid search (vector + BM25 keyword) with semantic ranker on Azure AI Search,
           using DefaultAzureCredential."
  Deliverable: This file — runnable Python script

Source: https://github.com/microsoft/skills/tree/main/.github/plugins/azure-sdk-python/skills/azure-search-documents-py
"""
import os
from azure.identity import DefaultAzureCredential
from azure.search.documents import SearchClient
from azure.search.documents.models import VectorizedQuery

endpoint = os.environ["SEARCH_ENDPOINT"]  # https://<service>.search.windows.net
index_name = os.environ.get("SEARCH_INDEX", "agent-docs")
credential = DefaultAzureCredential()

client = SearchClient(endpoint=endpoint, index_name=index_name, credential=credential)

# Hybrid search: vector + keyword + semantic ranker
query_text = "How does Azure MCP Server authenticate?"
query_vector = [0.1] * 1536  # Replace with real embedding from Azure OpenAI

results = client.search(
    search_text=query_text,
    vector_queries=[VectorizedQuery(vector=query_vector, k_nearest_neighbors=5, fields="embedding")],
    query_type="semantic",
    semantic_configuration_name="default",
    top=5,
    select=["title", "content", "url"],
)

print(f"=== Hybrid search: '{query_text}' ===")
for r in results:
    print(f"  [{r['@search.score']:.2f}] {r.get('title', 'N/A')}")
    print(f"    {r.get('content', '')[:120]}...")
    print(f"    Source: {r.get('url', 'N/A')}")

File: Agents/Azure-Agent-Skills-In-Action/skill-demos/sdk-typescript/05_azure_search_documents.ts
// TRIPLE:
//   Skill: azure-search-documents-ts
//   Prompt: "Using azure-search-documents-ts skill, write TypeScript code that performs hybrid search (vector+BM25) with semantic ranker."
//   Deliverable: This file
//   Source: https://github.com/microsoft/skills/tree/main/.github/plugins/azure-sdk-typescript/skills/azure-search-documents-ts

import { SearchClient, AzureKeyCredential } from "@azure/search-documents";
import { DefaultAzureCredential } from "@azure/identity";

async function main(): Promise<void> {
  const endpoint = process.env["AZURE_SEARCH_ENDPOINT"]!;
  const indexName = process.env["AZURE_SEARCH_INDEX"] ?? "skill-demo-index";
  const credential = new DefaultAzureCredential();

  const client = new SearchClient(endpoint, indexName, credential);

  // Hybrid search: keyword (BM25) + vector + semantic ranker
  const queryText = "How to deploy a model on Azure AI Foundry?";
  const queryVector: number[] = new Array(1536).fill(0).map(() => Math.random()); // placeholder embedding

  const results = await client.search(queryText, {
    vectorSearchOptions: {
      queries: [
        {
          kind: "vector",
          vector: queryVector,
          kNearestNeighborsCount: 5,
          fields: ["contentVector"],
        },
      ],
    },
    queryType: "semantic",
    semanticSearchOptions: {
      configurationName: "my-semantic-config",
    },
    top: 5,
    select: ["title", "content", "url"],
  });

  console.log("=== Hybrid Search Results ===");
  for await (const result of results.results) {
    const doc = result.document as any;
    console.log(`  Score: ${result.score?.toFixed(4)}`);
    console.log(`  Reranker: ${result.rerankerScore?.toFixed(4) ?? "N/A"}`);
    console.log(`  Title: ${doc.title}`);
    console.log("---");
  }
}

main().catch(console.error);
